common/compatibility/checker.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * This class is responsible for checking if the server's PHP version
+ * and the client's browser are compatible with OpenEMR. Please note
+ * that user agent sniffing is used to determine the browser. This is
+ * generally a bad practice, but we must ensure users are using only
+ * the officially supported/tested browsers.
+ *
+ * Copyright (C) 2017 Matthew Vita <matthewvita48@gmail.com>
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>.
+ *
+ * @package OpenEMR
+ * @author  Matthew Vita <matthewvita48@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+class Checker {
+    /**
+     * This is needed per the modernization project.
+     */
+    private $minimumPhpVersion = "5.5.9";
+
+    /**
+     * List of officially supported/tested browsers. Versions
+     * aren't specified because these browsers tend to self
+     * update/encourage end users to stay current.
+     */
+    private $supportedBrowsers = array("Firefox",
+                                       "Chrome");
+
+    /**
+     * Default constructor.
+     */
+    public function __construct() {}
+
+    /**
+     * Determines if a message should be sent to the user
+     * about their PHP / browser compatibility.
+     *
+     * @return empty string or string with compatability warnings.
+     */
+    public function getCompatibility() {
+        $phpCheck = $this->isPhpSupported();
+        $browserCheck = $this->isBrowserSupported();
+        $response = "";
+
+        if (!$phpCheck) {
+            $response .= "PHP version needs to be at least " . $this->minimumPhpVersion . ".";
+        }
+
+        if (!$browserCheck) {
+            if (!empty($response)) {
+                $response .= "\\n\\n";
+            }
+
+            $response .= "Your browser needs to be one of the following: ";
```

Comment:
In this case, check for the existence of the xlt() function. If it doesn't exist, then can create a dummy xlt() function on the fly that does nothing and  then run the message through the xlt() function.